docs/source/error-codes/self-service/index.rst

```diff
@@ -7,6 +7,203 @@ Error Codes
 .. toctree::
    :hidden:
 
+Overview
+*********
+
+
+.. _gRPC status codes: https://grpc.github.io/grpc/core/md_doc_statuscodes.html
+.. _gRPC status code: https://grpc.github.io/grpc/core/md_doc_statuscodes.html
+.. _StatusRuntimeException: https://grpc.github.io/grpc-java/javadoc/io/grpc/StatusRuntimeException.html
+.. _rich gRPC error model: https://cloud.google.com/apis/design/errors#error_details
+.. _standard gRPC description: https://grpc.github.io/grpc-java/javadoc/io/grpc/Status.html#getDescription--
+
+
+The majority of the errors are a result of some request processing.
+They are logged and returned to the user as a failed gRPC request
+using the standard StatusRuntimeException_.
+As such this approach remains unchanged in principle while self-service error codes aim at
+enhancing it by providing:
+
+- improved consistency of the returned errors across API endpoints,
+
+- richer error payload format with clearly distinguished machine readable parts to facilitate
+  automated error handling strategies,
+
+- complete inventory of all error codes with an explanation, suggested resolution and
+  other useful information.
+
+
+The goal is to enable users, developers and operators to act on the encountered
+errors in a self-service manner, either in an automated-way or manually.
+
+Feature Flag
+---------------------------
+
+You can enable self-service error-codes by specifying ``--use-self-service-error-codes``
+from command line.
+
+By default self-service error codes are turned off.
```

Comment:
It will be enabled by default. Let's not change this in this PR, though. I'll do it in a subsequent PR